eng/common/scripts/Package-Properties.ps1

```diff
@@ -164,21 +175,37 @@ function Get-CSVMetadata ([string]$MetadataUri=$MetadataUri)
     return $metadataResponse
 }
 
-function Operate-OnPackages ($activePkgList, $ServiceDirectory, [Array]$pkgPropsResult)
+function Operate-OnArtifacts ($activeArtifactList, $serviceDirectoryPath, $serviceDirectoryName, [Array]$pkgPropsResult)
 {
-    foreach ($pkg in $activePkgList)
+    foreach ($artifact in $activeArtifactList)
     {
-        LogDebug "Operating on $($pkg["name"])"
-        $pkgProps = Get-PkgProperties -PackageName $pkg["name"] -ServiceDirectory $ServiceDirectory
-        if ($null -ne  $pkgProps)
+        $directoriesPresent = Get-ChildItem $serviceDirectoryPath -Directory
+        $artifactName = $artifact["name"]
+
+        foreach ($directory in $directoriesPresent)
```

Comment:
I think it might make more sense to loop over the directory and open all the project files once and then build a list based on that and then find the ones that are applicable to artifact list. 

So perhaps we have a function that essentially parses the project for all the directories in a service directory and then we post process those to try and align the artifact names to those packages.